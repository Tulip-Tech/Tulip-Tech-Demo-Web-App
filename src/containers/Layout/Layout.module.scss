@use 'src/styles/variables';
@use 'src/styles/theme';
@use 'src/styles/mixins/responsive';
.layout {
    display: flex;
    flex-grow: 1;
    flex-direction: column;
}

.main {
    flex: 1;
}

div.footer {
    padding: 4% 1% 0% 1%;
    line-height: 18px;
    letter-spacing: 0.15px;
    text-align: center;
    text-shadow: 0 2px 4px rgba(0, 0, 0, 0.14), 0 3px 4px rgba(0, 0, 0, 0.12), 0 1px 5px rgba(0, 0, 0, 0.2);

    > div > a,
    a:visited,
    a:active,
    a:hover {
        color: variables.$white;
        text-decoration: none;
    }
}

.divider {
    width: 100%;
    border: none;
    border-top: 1px solid rgba(variables.$white, 0.12);
}

.buttonContainer {
    display: flex;
    flex-direction: column;
    padding: 0 variables.$base-spacing;

    > button:first-child {
        margin-bottom: calc(#{variables.$base-spacing} / 2);
    }
}
.footerContent {
    display: flex;
    width: 40%;
    padding: 5% 2% 0% 2%;
    @include responsive.mobile-only() {
        width: 90%;
        font-size: 10px;
    }

    @include responsive.tablet-only() {
        width: 60%;
        font-size: 10px;
    }
}

.link {
    padding: 3% 7%;
    color: variables.$white;
    font-family: var(--body-alt-font-family);
}

.linkSection {
    display: grid;
    width: 33.33%;
}
.bookbtn {
    color: variables.$white;
}
.Header {
    display: flex;
}
.headerShopButton {
    padding: 0 20px;
    color: variables.$white;
    font-weight: 600;
    font-size: 14px;
    text-decoration: none;
    background-color: var(--highlight-color, #29abe2);
}
.headerShopButton:hover {
    padding: 0 20px;
    color: variables.$white;
    font-weight: 600;
    font-size: 14px;
    text-decoration: underline;
    background-color: var(--highlight-color, #29abe2);
}
