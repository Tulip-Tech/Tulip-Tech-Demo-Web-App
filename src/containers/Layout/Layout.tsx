import React, { useEffect, useRef, useState } from 'react';
import { Helmet } from 'react-helmet';
import { useTranslation } from 'react-i18next';
import { Outlet, useLocation, useNavigate } from 'react-router';
import shallow from 'zustand/shallow';

import styles from './Layout.module.scss';

import { useAccountStore } from '#src/stores/AccountStore';
import { useUIStore } from '#src/stores/UIStore';
import { useConfigStore } from '#src/stores/ConfigStore';
import useSearchQueryUpdater from '#src/hooks/useSearchQueryUpdater';
import useClientIntegration from '#src/hooks/useClientIntegration';
import Button from '#components/Button/Button';
import MarkdownComponent from '#components/MarkdownComponent/MarkdownComponent';
import Header from '#components/Header/Header';
import Sidebar from '#components/Sidebar/Sidebar';
import MenuButton from '#components/MenuButton/MenuButton';
import UserMenu from '#components/UserMenu/UserMenu';
import { addQueryParam } from '#src/utils/location';
import Link from '#src/components/Link/Link';

const Layout = () => {
    const location = useLocation();
    const navigate = useNavigate();
    const { t } = useTranslation('common');
    const { config, accessModel } = useConfigStore(({ config, accessModel }) => ({ config, accessModel }), shallow);
    const { menu, assets, siteName, description, styling, features } = config;
    const { clientId } = useClientIntegration();
    const { searchPlaylist } = features || {};
    const { footerText } = styling || {};

    const { searchQuery, searchActive, userMenuOpen } = useUIStore(
        ({ searchQuery, searchActive, userMenuOpen }) => ({
            searchQuery,
            searchActive,
            userMenuOpen,
        }),
        shallow,
    );
    const { updateSearchQuery, resetSearchQuery } = useSearchQueryUpdater();
    const isLoggedIn = !!useAccountStore((state) => state.user);

    const searchInputRef = useRef<HTMLInputElement>(null) as React.MutableRefObject<HTMLInputElement>;

    const [sideBarOpen, setSideBarOpen] = useState(false);
    const banner = assets.banner;

    useEffect(() => {
        if (searchActive && searchInputRef.current) {
            searchInputRef.current.focus();
        }
    }, [searchActive]);

    const searchButtonClickHandler = () => {
        useUIStore.setState({
            searchActive: true,
            preSearchPage: location,
        });
    };

    const closeSearchButtonClickHandler = () => {
        resetSearchQuery();

        useUIStore.setState({
            searchActive: false,
        });
    };

    const loginButtonClickHandler = () => {
        navigate(addQueryParam(location, 'u', 'login'));
    };

    const signUpButtonClickHandler = () => {
        navigate(addQueryParam(location, 'u', 'create-account'));
    };

    const toggleUserMenu = (value: boolean) =>
        useUIStore.setState({
            userMenuOpen: value,
        });

    const renderUserActions = () => {
        if (!clientId) return null;

        return isLoggedIn ? (
            <UserMenu showPaymentsItem={accessModel !== 'AVOD'} />
        ) : (
            <div className={styles.buttonContainer}>
                <Button fullWidth onClick={loginButtonClickHandler} label={t('sign_in')} />
                <Button variant="contained" color="primary" onClick={signUpButtonClickHandler} label={t('sign_up')} fullWidth />
            </div>
        );
    };

    return (
        <div className={styles.layout}>
            <Helmet>
                <title>{siteName}</title>
                <meta name="description" content={description} />
                <meta property="og:description" content={description} />
                <meta property="og:title" content={siteName} />
                <meta name="twitter:title" content={siteName} />
                <meta name="twitter:description" content={description} />
            </Helmet>
            <div className={styles.main}>
                <Header
                    onMenuButtonClick={() => setSideBarOpen(true)}
                    logoSrc={banner}
                    searchEnabled={!!searchPlaylist}
                    searchBarProps={{
                        query: searchQuery,
                        onQueryChange: (event) => updateSearchQuery(event.target.value),
                        onClearButtonClick: () => updateSearchQuery(''),
                        inputRef: searchInputRef,
                    }}
                    searchActive={searchActive}
                    onSearchButtonClick={searchButtonClickHandler}
                    onCloseSearchButtonClick={closeSearchButtonClickHandler}
                    onLoginButtonClick={loginButtonClickHandler}
                    onSignUpButtonClick={signUpButtonClickHandler}
                    isLoggedIn={isLoggedIn}
                    userMenuOpen={userMenuOpen}
                    toggleUserMenu={toggleUserMenu}
                    canLogin={!!clientId}
                    showPaymentsMenuItem={accessModel !== 'AVOD'}
                >
                    <div className={styles.Header}>
                        <Button label={t('home')} to="/home" variant="text" />
                        {menu.map((item) => (
                            <Button key={item.contentId} label={item.label} to={`/p/${item.contentId}`} variant="text" />
                        ))}
                        <Button label={t('Book Livestreams')} to="/book-livestream" color="headerbtn" />
                        <Button label={t('Book Player Highlights')} to="/book-player-highlights" color="headerbtn"></Button>
                        <button className={styles.headerShopButton}>
                            <a href="https://shop.gameinframe.com/" className={styles.headerShopButton}>
                                Shop
                            </a>
                        </button>
                    </div>
                </Header>
                <Sidebar isOpen={sideBarOpen} onClose={() => setSideBarOpen(false)}>
                    <MenuButton label={t('home')} to="/" tabIndex={sideBarOpen ? 0 : -1} />
                    {menu.map((item) => (
                        <MenuButton key={item.contentId} label={item.label} to={`/p/${item.contentId}`} tabIndex={sideBarOpen ? 0 : -1} />
                    ))}
                    <Button className={styles.headerbtn} label={t('Book Livestreams')} to="/book-livestream" color="headerbtn" />
                    <Button label={t('Book Player Highlights')} to="/book-player-highlights" color="headerbtn"></Button>
                    {/* <Button label={t('Shop')} to="/highlights" color="headerbtn"></Button> */}
                    {/* <button className={styles.headerShopButton}>
                        <a href="https://shop.gameinframe.com/" className={styles.headerShopButton}>
                            Shop
                        </a>
                    </button> */}
                    <hr className={styles.divider} />
                    {renderUserActions()}
                </Sidebar>
                <Outlet />
            </div>
            {!!footerText && (
                <div>
                    <div className={styles.footerContent}>
                        <div className={styles.linkSection}>
                            <Link to="/Advertise" className={styles.link}>
                                Advertise
                            </Link>

                            <Link to="/Abouts" className={styles.link}>
                                About us
                            </Link>

                            <Link to="/Newletters" className={styles.link}>
                                Newsletters
                            </Link>
                        </div>
                        <div className={styles.linkSection}>
                            <Link to="/Contacts" className={styles.link}>
                                Contact us
                            </Link>

                            <Link to="/Job" className={styles.link}>
                                Jobs
                            </Link>
                        </div>
                        <div className={styles.linkSection}>
                            <Link to="/privacy-policy" className={styles.link}>
                                Privacy Policy
                            </Link>
                            <Link to="/terms" className={styles.link}>
                                Terms of Use
                            </Link>
                        </div>
                    </div>
                    <MarkdownComponent className={styles.footer} markdownString={footerText} inline />
                </div>
            )}

            {/* <Link to="/privacy-policy">Privacy-Policy</Link> */}
        </div>
    );
};

export default Layout;
