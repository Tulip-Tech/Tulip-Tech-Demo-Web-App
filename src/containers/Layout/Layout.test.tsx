import React from 'react';

import Layout from './Layout';

import { renderWithRouter } from '#test/testUtils';

describe('<Layout />', () => {
    test('renders layout', () => {
        const { container } = renderWithRouter(<Layout />);

        expect(container).toMatchSnapshot();
    });
});
