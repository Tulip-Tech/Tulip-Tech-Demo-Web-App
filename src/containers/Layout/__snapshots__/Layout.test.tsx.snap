// Vitest Snapshot v1

exports[`<Layout /> > renders layout 1`] = `
<div>
  <div
    class="_layout_c71437"
  >
    <div
      class="_main_c71437"
    >
      <header
        class="_header_f4f7a7 _static_f4f7a7"
      >
        <div
          class="_container_f4f7a7"
        >
          <div
            class="_menu_f4f7a7"
          >
            <div
              aria-label="open_menu"
              class="_iconButton_0fef65 _iconButton_f4f7a7"
              role="button"
              tabindex="0"
            >
              <svg
                aria-hidden="true"
                class="_icon_1e9999"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <nav
            aria-label="menu"
            class="_nav_f4f7a7"
          >
            <a
              aria-current="page"
              class="_button_f8f296 _default_f8f296 _text_f8f296 _active_f8f296"
              href="/"
            >
              <span>
                home
              </span>
            </a>
          </nav>
          <div
            class="_search_f4f7a7"
          />
        </div>
      </header>
      <div
        class="_backdrop_577f70"
      />
      <div
        class="_sidebar_577f70"
      >
        <div
          class="_heading_577f70"
        >
          <div
            aria-label="close_menu"
            class="_iconButton_0fef65"
            role="button"
            tabindex="-1"
          >
            <svg
              aria-hidden="true"
              class="_icon_1e9999"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </g>
            </svg>
          </div>
        </div>
        <nav
          class="_group_577f70"
        >
          <a
            aria-current="page"
            aria-label="home"
            class="_menuButton_91706b _active_91706b"
            href="/"
            tabindex="-1"
          >
            <span>
              home
            </span>
          </a>
          <hr
            class="_divider_c71437"
          />
        </nav>
      </div>
    </div>
  </div>
</div>
`;
